src/custom/hooks/useApproveCallback/index.ts

```diff
@@ -29,22 +37,37 @@ export function useApproveCallbackFromTrade(
 
   const vaultRelayer = chainId ? GP_VAULT_RELAYER[chainId] : undefined
 
-  return useApproveCallback(openTransactionConfirmationModal, closeModals, amountToApprove, vaultRelayer)
+  return useApproveCallback({
+    openTransactionConfirmationModal,
+    closeModals,
+    amountToApprove,
+    spender: vaultRelayer,
+    amountToCheckAgainstAllowance,
+  })
 }
 
 export type OptionalApproveCallbackParams = {
   transactionSummary: string
 }
 
-export function useApproveCallbackFromClaim(
-  openTransactionConfirmationModal: (message: string) => void,
-  closeModals: () => void,
-  amountToApprove: CurrencyAmount<Currency> | undefined
-) {
+type ApproveCallbackFromClaimParams = Omit<ApproveCallbackParams, 'spender'>
+
+export function useApproveCallbackFromClaim({
+  openTransactionConfirmationModal,
+  closeModals,
+  amountToApprove,
+  amountToCheckAgainstAllowance,
+}: ApproveCallbackFromClaimParams) {
   const { chainId } = useActiveWeb3React()
 
   const vCowContract = chainId ? V_COW_CONTRACT_ADDRESS[chainId] : undefined
 
   // Params: modal cbs, amountToApprove: token user is investing e.g, spender: vcow token contract
-  return useApproveCallback(openTransactionConfirmationModal, closeModals, amountToApprove, vCowContract)
+  return useApproveCallback({
+    openTransactionConfirmationModal,
+    closeModals,
+    amountToApprove,
+    spender: vCowContract,
```

Comment:
I see this is not from this PR but maybe it's better we don't name even internally `spender` as `vCowContract` because this hook could be used somewhere else?

Reply:
good point, although the hook is called `useApproveCallbackFromClaim`, so i think is not a big deal